src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -162,8 +163,77 @@ public ReplicationLoader(DocumentDatabase database, ServerStore server)
                 null, reconnectTime, reconnectTime);
             MinimalHeartbeatInterval = (int)config.ReplicationMinimalHeartbeat.AsTimeSpan.TotalMilliseconds;
             database.TombstoneCleaner.Subscribe(this);
+            server.Cluster.Changes.DatabaseChanged+=DatabaseValueChanged;
         }
 
+        
+       private Task DatabaseValueChanged(string databaseName, long index, string type, DatabasesLandlord.ClusterDatabaseChangeType changeType, object changeState)
+       {
+           var documentDatabase = Database;
+           if (documentDatabase == null)
+               return Task.CompletedTask;
+           
+           if (string.Equals(documentDatabase.Name, databaseName, StringComparison.OrdinalIgnoreCase) == false)
+               return Task.CompletedTask;
+
+           switch (changeState)
+           {
+               case UpdatePullReplicationAsHubCommand put:
+                   DisposeRelatedPullReplication(put.Definition.Name, null /*all*/);
```

Comment:
So if I'm adding a new cert to the list, this will break all running connections?  not sure it is nice on a large scale

Reply:
This is when you update the _hub_ itself, not adding / removing / updating a cert